allennlp/common/tensor.py

```diff
@@ -123,3 +123,68 @@ def masked_softmax(vector, mask):
     else:
         # There is no mask, so we use the provided ``torch.nn.functional.softmax`` function.
         return torch.nn.functional.softmax(vector)
+
+
+def get_text_field_mask(text_field_tensors: Dict[str, torch.Tensor]) -> torch.Tensor:
```

Comment:
I wanted it to be a function so that the model code doesn't have to think about the content of the text field arrays at all.  For higher-order tensors, I was thinking of things like memory network inputs, sets of paragraphs for TriviaQA, or other places you have a `ListField[TextField]`.  To handle this and remove the magical "tokens" key, I think the right thing to do is just find the entry in the dictionary that has the minimum number of dimensions, and use that.